ipc-agent-sdk/src/apis/topdown.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+
+use crate::apis::{json_rpc_methods, IpcAgentClient};
+use crate::jsonrpc::JsonRpcClient;
+use crate::message::chain::ChainHeadResponse;
+use fvm_shared::clock::ChainEpoch;
+use ipc_sdk::cross::CrossMsg;
+use ipc_sdk::subnet_id::SubnetID;
+use ipc_sdk::ValidatorSet;
+
+impl<T: JsonRpcClient> IpcAgentClient<T> {
+    pub async fn get_top_down_msgs(
+        &self,
+        subnet_id: &SubnetID,
+        start: ChainEpoch,
+        to: ChainEpoch,
+    ) -> anyhow::Result<Vec<CrossMsg>> {
+        self.json_rpc_client
+            .request::<Vec<CrossMsg>>(
+                json_rpc_methods::LIST_TOPDOWN_MSGS,
+                serde_json::json!({
+                    "subnet_id": subnet_id.to_string(),
+                    "start": start,
+                    "to": to,
+                }),
+            )
+            .await
+    }
+
+    pub async fn get_chain_head(&self, subnet_id: String) -> anyhow::Result<ChainHeadResponse> {
```

Comment:
The other day I was chatting with Jorge, and for those functions that directly interact with the subnet without requiring anything from the agent we should consider moving the RPC implementation to the SDK and make the all directly. That way we can use the `ipc-agent` without requiring a `daemon` running. I think this could really improve the UX.

Reply:
Added